src/OptimalTransport.jl

```diff
@@ -125,14 +125,17 @@ The Gibbs kernel `K` is given by `K = exp.(- C / eps)` where `C` is the cost mat
 `eps` the entropic regularization parameter. The optimal transport plan for histograms `u`
 and `v` and cost matrix `C` with regularization parameter `eps` can be computed as
 `Diagonal(u) * K * Diagonal(v)`.
+
+Note that for a common kernel `K`, multiple histograms may be provided for a batch computation by passing `mu` and `nu`
+as matrices whose columns `mu[:, i]` and `nu[:, i]` correspond to pairs of histograms. 
+The output are then matrices `u` and `v` such that `u[:, i]` and `v[:, i]` are the dual variables for `mu[:, i]` and `nu[:, i]`.
 """
 function sinkhorn_gibbs(mu, nu, K; tol=1e-9, check_marginal_step=10, maxiter=1000)
-    if !(sum(mu) ≈ sum(nu))
+    if !(sum(mu; dims=1) ≈ sum(nu; dims=1))
         throw(ArgumentError("Error: mu and nu must lie in the simplex"))
     end
-
     # initial iteration
-    temp_v = vec(sum(K; dims=2))
+    temp_v = repeat(vec(sum(K; dims=2)); outer=(1, size(mu, 2)))
```

Comment:
I assume this can be simplified by changing the algorithm to
```julia
u = mu ./ vec(sum(K; dims=2))
temp_u = K' * u
v = nu ./ temp_u
temp_v = K * v
```
and adjusting the `for` loop accordingly (i.e. move the `mul!(temp_v, K, v)` to the end of the computations). Then `temp_v` (and the other variable) should be of the correct shape automatically.